runtime/src/bank/fee_distribution.rs

```diff
@@ -72,37 +83,43 @@ impl Bank {
             fee_budget_limits.prioritization_fee,
             FeeFeatures::from(self.feature_set.as_ref()),
         );
-        let (reward, _burn) =
-            self.calculate_reward_and_burn_fee_details(&CollectorFeeDetails::from(fee_details));
+        let FeeDistribution {
+            deposit: reward,
+            burn: _,
+        } = self.calculate_reward_and_burn_fee_details(&CollectorFeeDetails::from(fee_details));
         reward
     }
 
     pub fn calculate_reward_and_burn_fee_details(
         &self,
         fee_details: &CollectorFeeDetails,
-    ) -> (u64, u64) {
-        let (deposit, burn) = if fee_details.transaction_fee != 0 {
-            self.burn_transaction_fee(fee_details.transaction_fee)
-        } else {
-            (0, 0)
-        };
-        (deposit.saturating_add(fee_details.priority_fee), burn)
-    }
+    ) -> FeeDistribution {
+        if fee_details.transaction_fee == 0 {
+            return FeeDistribution::default();
+        }
 
-    fn burn_transaction_fee(&self, fees: u64) -> (u64, u64) {
-        let burned = fees * self.burn_percent() / 100;
-        (fees - burned, burned)
+        let burn = fee_details.transaction_fee * self.burn_percent() / 100;
+        let deposit = fee_details
+            .priority_fee
+            .saturating_add(fee_details.transaction_fee.saturating_sub(burn));
+        FeeDistribution { deposit, burn }
     }
 
     const fn burn_percent(&self) -> u64 {
         // NOTE: burn percent is statically 50%, in case it needs to change in the future,
         // burn_percent can be bank property that being passed down from bank to bank, without
         // needing fee-rate-governor
+        static_assertions::const_assert!(solana_sdk::fee_calculator::DEFAULT_BURN_PERCENT <= 100);
+
         solana_sdk::fee_calculator::DEFAULT_BURN_PERCENT as u64
     }
 
-    fn deposit_or_burn_fee(&self, deposit: u64, burn: &mut u64) {
-        match self.deposit_fees(&self.collector_id, deposit) {
+    fn deposit_or_burn_fee(&self, deposit: u64) -> u64 {
```

Comment:
```suggestion
    /// Attempts to deposit the given `deposit` amount into the fee collector account.
    /// 
    /// Returns the original `deposit` amount if the deposit failed and must be burned, otherwise 0.
    fn deposit_or_burn_fee(&self, deposit: u64) -> u64 {
```